File: dotnet/ApkResouceParser.NetStandard2/XmlNamespaceChunk.cs
namespace ApkResourceParser
{
    /// <summary>
    /// Represents the start/end of a namespace in an XML document.
    /// </summary>
    public abstract class XmlNamespaceChunk : XmlNodeChunk
    {
        /// <summary>
        /// A string reference to the namespace prefix.
        /// </summary>
        private readonly int _prefix;

        /// <summary>
        /// A string reference to the namespace URI.
        /// </summary>
        private readonly int _uri;

        protected internal XmlNamespaceChunk(ByteBuffer buffer, Chunk parent) : base(buffer, parent)
        {
            _prefix = buffer.getInt();
            _uri = buffer.getInt();
        }

        /// <summary>
        /// Returns the namespace prefix.
        /// </summary>
        public virtual string getPrefix()
        {
            return getString(_prefix);
        }

        /// <summary>
        /// Returns the namespace URI.
        /// </summary>
        public virtual string getUri()
        {
            return getString(_uri);
        }


        /// <summary>
        /// Returns a brief description of this namespace chunk. The representation of this information is
        /// subject to change, but below is a typical example:
        /// 
        /// <pre>
        /// "XmlNamespaceChunk{line=1234, comment=My awesome comment., prefix=foo, uri=com.google.foo}"
        /// </pre>
        /// </summary>
        public override string ToString()
        {
            return $"XmlNamespaceChunk{{line={getLineNumber()}, comment={getComment()}, prefix={getPrefix()}, uri={getUri()}}}";
        }
    }
}

File: java/apk-resource-parser/src/main/java/ikas/project/java/apk/arsc/XmlNamespaceChunk.java
package ikas.project.java.apk.arsc;

import java.nio.ByteBuffer;
import java.util.Locale;

/**
 * Represents the start/end of a namespace in an XML document.
 */
public abstract class XmlNamespaceChunk extends XmlNodeChunk {

    /**
     * A string reference to the namespace prefix.
     */
    private final int prefix;

    /**
     * A string reference to the namespace URI.
     */
    private final int uri;

    protected XmlNamespaceChunk(ByteBuffer buffer, /*@Nullable*/ Chunk parent) {
        super(buffer, parent);
        prefix = buffer.getInt();
        uri = buffer.getInt();
    }

    /**
     * Returns the namespace prefix.
     */
    public String getPrefix() {
        return getString(prefix);
    }

    /**
     * Returns the namespace URI.
     */
    public String getUri() {
        return getString(uri);
    }


    /**
     * Returns a brief description of this namespace chunk. The representation of this information is
     * subject to change, but below is a typical example:
     *
     * <pre>
     * "XmlNamespaceChunk{line=1234, comment=My awesome comment., prefix=foo, uri=com.google.foo}"
     * </pre>
     */
    @Override
    public String toString() {
        return String.format(
                Locale.US,
                "XmlNamespaceChunk{line=%d, comment=%s, prefix=%s, uri=%s}",
                getLineNumber(),
                getComment(),
                getPrefix(),
                getUri());
    }
}
